docs/src/components/code-examples.test.js

```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import renderer from 'react-test-renderer'
+import { filterTransformSortCodeSnippets } from './code-examples'
+import mockGraphQLResponse from './code-examples.test.fixture'
+
+const mockData = mockGraphQLResponse.allCodeSamples.edges
+
+test.each`
+  data         | path                | expectedLength | testDescription
+  ${mockData}  | ${'sending-emails'} | ${3}           | ${'filters correctly'}
+  ${mockData}  | ${'unknown-path'}   | ${0}           | ${'returns empty array no matches found'}
+  ${[]}        | ${'sending-emails'} | ${0}           | ${'returns empty array when data is empty'}
+  ${null}      | ${'sending-emails'} | ${0}           | ${'returns empty array when data is null'}
+  ${undefined} | ${'sending-emails'} | ${0}           | ${'returns empty array when data is undefined'}
```

Comment:
Regarding `null` and `undefined` as passed to this func, do you know what graphql actually passes if there are no results? Because if it's undefined you can change the actual signature of the function to just:

```js
export const filterTransformSortCodeSnippets = ({ codeSnippets = [], path }) => {
```
And remove the check at the start. If graphql passes `null` then unfortunately we need to keep it in there.